src/DynamoCoreWpf/Extensions/ViewLoadedParams.cs

```diff
@@ -26,36 +29,44 @@ public class ViewLoadedParams
         // Here's a start on the implementation
         //
 
-        //private readonly DynamoView view;
-
-        //internal ViewLoadedParams(DynamoView view)
-        //{
-        //    this.view = view;
-        //}
-
-        ///// <summary>
-        ///// Add a menu item for workspace context click
-        ///// </summary>
-        ///// <param name="item">The item to insert</param>
-        ///// <param name="options">Options object to determine in which cases the MenuItem should be visible</param>
-        //public void AddWorkspaceContextClickMenuItem(MenuItem item, WorkspaceContextClickOptions options)
-        //{
-            
-        //}
-
-        //public void AddShortcutBarItem(ShortcutBarItem item)
-        //{
-        //    // add an item to Dynamo's shortcut bar
-        //}
-
-        //public void AddMenuBar(MenuItem item)
-        //{
-            
-        //}
-
-        //public void AddKeyBinding(KeyBinding binding)
-        //{
-            
-        //}
+        private DynamoView dynamoView;
+        private DynamoViewModel dynamoViewModel;
+
+        public IEnumerable<IWorkspaceModel> WorkSpaces
+        {
+            get
+            {
+                return dynamoViewModel.Model.Workspaces;
+            }
+        }
+
+        public ViewLoadedParams(DynamoView dynamoV, DynamoViewModel dynamoVM)
+        {
+            dynamoView = dynamoV;
+            dynamoViewModel = dynamoVM;
+        }
+
+        public void AddMenuItem(MenuBarType type, MenuItem menuItem)
+        {
+            var dynamoMenu = dynamoView.titleBar.ChildOfType<Menu>();
+
+            if (dynamoMenu == null)
+                return;
+
+            var dynamoMenuItems = dynamoMenu.Items.OfType<MenuItem>();
+            var dynamoItem = dynamoMenuItems.First(item => item.Header.ToString() == "_" + type);
+            if (dynamoItem == null)
+                return;
+
+            dynamoItem.Items.Add(menuItem);
```

Comment:
Here we should keep track of the added `MenuItem`'s.  When we dispose of the `ViewExtension`, we should remove  the `MenuItem`.